pkg/logger/README.md

```diff
@@ -148,15 +265,66 @@ logger.Errorf("Failed to connect to %s: %v", host, err)
 4. **Resource Cleanup**
    ```go
    defer logger.Sync()
+   
+   // For OpenTelemetry integration
+   defer func() {
+       ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+       defer cancel()
+       logger.ShutdownOpenTelemetryLogger(ctx)
+   }()
    ```
 
+5. **OpenTelemetry Best Practices**
+   - Always set a meaningful `ServiceName` for better observability
+   - Use appropriate protocol (`http` vs `grpc`) based on your infrastructure
+   - Include authentication headers when required by your observability platform
+   - Handle shutdown gracefully to ensure all logs are flushed
+   - Test connectivity to your OpenTelemetry endpoint before production deployment
+
+## Configuration Examples
+
+### Development Configuration
+
+```yaml
+# config.yaml
+logger:
+  level: "debug"
+  encoding: "console"
+  opentelemetry:
+    enabled: true
+    service_name: "my-service-dev"
+    endpoint: "http://localhost:4318"
+    protocol: "http"
+    insecure: true
+```
+
+### Production Configuration
+
+```yaml
+# config.yaml
+logger:
+  level: "info"
+  encoding: "json"
+  opentelemetry:
+    enabled: true
+    service_name: "my-service-prod"
+    endpoint: "https://otel-collector.company.com:4318"
+    protocol: "http"
+    insecure: false
+    headers:
+      authorization: "Bearer ${OTEL_TOKEN}"
+      x-api-key: "${API_KEY}"
```

Comment:
Is this an env var? It seems a bit generic. Did we follow the golem way, where this can be loaded from env vars? Thinking about the api key set in a config map.